drivers/modem/ublox-sara-n310.c

```diff
@@ -0,0 +1,1069 @@
+/*
+ * Copyright (c) 2021 Abel Sensors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+#include <net/net_if.h>
+#include <net/net_offload.h>
+#include <net/socket_offload.h>
+
+#include <kernel.h>
+#include <errno.h>
+#include <zephyr.h>
+
+#include <drivers/modem/ublox-sara-n310.h>
+
+#include "modem_context.h"
+#include "modem_iface_uart.h"
+#include "modem_cmd_handler.h"
+#include "modem_socket.h"
+
+#define MDM_UART_DEV_NAME DT_INST_BUS_LABEL(0)
+#define MDM_MAX_DATA_LENGTH 512
+#define MDM_RECV_MAX_BUF 10
+#define MDM_RECV_BUF_SIZE 64
+#define MDM_CMD_TIMEOUT K_SECONDS(6)
+#define MDM_REGISTRATION_TIMEOUT K_SECONDS(20)
+
+#define RX_PRIORITY 5
+
+#define DT_DRV_COMPAT ublox_sara_n310
+LOG_MODULE_REGISTER(modem_ublox_sara_n310, CONFIG_MODEM_LOG_LEVEL);
+#define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
+
+/* modem info */
+struct modem_info {
+	char mdm_iccid[32];
+	char mdm_model[16];
+	char mdm_manufacturer[16];
+	char mdm_revision[32];
+	char mdm_imei[32];
+
+	char mdm_ip[32];
+};
+
+static struct modem_info minfo;
+
+/* modem_info getters */
+char *n310_get_model(void)
+{
+	return minfo.mdm_model;
+}
+
+char *n310_get_iccid(void)
+{
+	return minfo.mdm_iccid;
+}
+
+char *n310_get_manufacturer(void)
+{
+	return minfo.mdm_manufacturer;
+}
+
+char *n310_get_revision(void)
+{
+	return minfo.mdm_revision;
+}
+
+char *n310_get_imei(void)
+{
+	return minfo.mdm_imei;
+}
+
+char *n310_get_ip(void)
+{
+	return minfo.mdm_ip;
+}
+
+/* data struct */
+struct modem_data {
+	struct net_if *net_iface;
+	uint8_t mac_addr[6];
+
+	struct modem_context context;
+	struct modem_cmd_handler_data cmd_handler_data;
+	struct modem_iface_uart_data mdm_data;
+
+	uint8_t cmd_match_buf[128];
+	char mdm_rx_rb_buf[1024];
+
+	struct k_sem sem_response;
+	enum n310_network_state network_state;
+
+	/* socket data */
+	struct modem_socket_config socket_config;
+	struct modem_socket sockets[6];
+	int sock_written;
+};
+
+static struct modem_data mdata;
+
+/* socket read callback data */
+struct socket_read_data {
+	char *recv_buf;
+	size_t recv_buf_len;
+	struct sockaddr *recv_addr;
+	uint16_t recv_read_len;
+};
+
+/* rx thread data */
+NET_BUF_POOL_DEFINE(mdm_recv_pool, MDM_RECV_MAX_BUF, MDM_RECV_BUF_SIZE, 0,
+		    NULL);
+K_THREAD_STACK_DEFINE(modem_rx_stack, MDM_MAX_DATA_LENGTH);
+static struct k_thread rx_data;
+
+/* network state getter */
+int n310_get_state(void)
+{
+	int ret;
+
+	ret = modem_cmd_send(&mdata.context.iface, &mdata.context.cmd_handler,
+			     NULL, 0U, "AT+CEREG?", &mdata.sem_response,
+			     MDM_CMD_TIMEOUT);
+
+	if (ret < 0) {
+		return ret;
+	}
+
+	return mdata.network_state;
+}
+
+/* helper macro to keep readability */
+#define ATOI(s_, value_, desc_) modem_atoi(s_, value_, desc_, __func__)
```

Comment:
If you define your macros always put parentheses around your parameters
E.g. 
`
#define ATOI(s_, value_, desc_) modem_atoi((s_), (value_), (desc_), (__func__))
`